### PR TITLE
Rotation

### PR DESCRIPTION
**Pull request by Eignar17**
to http://cgit.compiz.org/~satamaxx/earth/commit/?id=b0cf3ab7452848569e5002b9df1266e85f51a995
author	Maxime Wack maximewack@free.fr

### CODE REVIEW

File: src/earth/earth.c [hunk 1]
```diff
@@ -496,6 +496,9 @@ earthScreenOptionChanged (CompScreen		*s,
 	case EarthScreenOptionClouds:
 		es->clouds = earthGetClouds (s);
 	break;
+	case EarthScreenOptionRotation:
+		es->rotspeed    = earthGetRotation (s);
```
Comment (soreau) [inline src/earth/earth.c]: This should be 
`es->rotspeed = earthGetRotation (s);`

File: src/earth/earth.c [hunk 2]
```diff
@@ -546,7 +549,14 @@ earthPreparePaintScreen (CompScreen *s,
 	es->gha = (float)currenttime->tm_hour-(es->tz +
 				(float)currenttime->tm_isdst) +
 				(float)currenttime->tm_min / 60.0f;
-
+				(float)currenttime->tm_sec/3600.0000f;
+	
+	/* Animation */
+	es->rotation += es->rotspeed;
+	if (es->rotation > 360)
+	es->rotation -= 360;
```
Comment (soreau) [inline src/earth/earth.c]: there should be an indent after if conditional like this:
```
if (es->rotation > 360)
	es->rotation -= 360;
```

File: src/earth/earth.c [hunk 3]
```diff
@@ -546,7 +549,14 @@ earthPreparePaintScreen (CompScreen *s,
...
+	else if (es->rotation < -360)
+	es->rotation += 360;
```
Comment (soreau) [inline src/earth/earth.c]: Indent here too.

File: src/earth/earth.c [hunk 4]
```diff
@@ -620,6 +630,12 @@ earthPaintInside (CompScreen			  *s,
 	glScalef (ratio * es->earth_size, es->earth_size, ratio * es->earth_size);
 	es->previousoutput = output->id;
 
+	/* Animation */
+	glRotatef (es->rotation, 0, 1, 0);
+    
+	if (cs->moMode == CUBE_MOMODE_MULTI)
+	glRotatef ((360.0f / s->nOutputDev) * output->id, 0, 1, 0);
```
Comment (soreau) [inline src/earth/earth.c]: Also indent here.

File: src/earth/earth.c [hunk 5]
```diff
@@ -723,7 +739,11 @@ earthClearTargetOutput (CompScreen *s,
 
 	/* Rotate the skydome according to the mouse and the rotation of the Earth */
 	glRotatef (vRotate - 90, 1.0f, 0.0f, 0.0f);
-	glRotatef (xRotate, 0.0f, 0.0f, 1.0f);
+	glRotatef (xRotate + es->rotation, 0.0f, 0.0f, 1.0f);
+    
+	if (cs->moMode == CUBE_MOMODE_MULTI)
+	glRotatef ((360.0f / s->nOutputDev) * currentoutput->id, 0, 0, 1);
```
Comment (soreau) [inline src/earth/earth.c]: Indent..

**Comment by Eignar17** [comment]
@soreau copy and paste code to http://cgit.compiz.org/~satamaxx/earth/commit/?id=b0cf3ab7452848569e5002b9df1266e85f51a995 :v:

**Comment by Eignar17** [comment]
@soreau update code

File: src/earth/earth.c [hunk 6]
```diff
@@ -788,6 +808,9 @@ earthInitScreen (CompPlugin *p,
 
 	es->damage = FALSE;
 
+	/* Animation */
+	es->rotation = 0;
```
Comment (soreau) [inline src/earth/earth.c]: In addition to this, you need to initialize `es->rotspeed` as well.

File: src/earth/earth.c [hunk 7]
```diff
@@ -546,7 +549,14 @@ earthPreparePaintScreen (CompScreen *s,
 	es->gha = (float)currenttime->tm_hour-(es->tz +
 				(float)currenttime->tm_isdst) +
 				(float)currenttime->tm_min / 60.0f;
-
+				(float)currenttime->tm_sec/3600.0000f;
+	
+	/* Animation */
+            es->rotation += es->rotspeed;
```
Comment (soreau) [inline src/earth/earth.c]: This is a problem. es->rotspeed is never initialized. See below where es->rotation is initialized.

File: src/earth/earth.c [hunk 8]
```diff
@@ -546,7 +549,14 @@ earthPreparePaintScreen (CompScreen *s,
 	es->gha = (float)currenttime->tm_hour-(es->tz +
 				(float)currenttime->tm_isdst) +
 				(float)currenttime->tm_min / 60.0f;
-
+				(float)currenttime->tm_sec/3600.0000f;
```
Comment (soreau) [inline src/earth/earth.c]: This statement does nothing. It was probably meant to replace `(float)currenttime->tm_min / 60.0f;` with `(float)currenttime->tm_sec/3600.0000f;`

File: src/earth/earth.c [hunk 9]
```diff
@@ -788,6 +807,11 @@ earthInitScreen (CompPlugin *p,
 
 	es->damage = FALSE;
 
+	/* Animation */
+	es->rotspeed
```
Comment (soreau) [inline src/earth/earth.c]: It should be `es->rotspeed = 0;`

Comment (Eignar17) [reply src/earth/earth.c]: ok ok I'm understanding

File: src/earth/earth.c [hunk 10]
```diff
@@ -545,8 +548,14 @@ earthPreparePaintScreen (CompScreen *s,
 				((float)currenttime->tm_yday + 10.0f));
 	es->gha = (float)currenttime->tm_hour-(es->tz +
 				(float)currenttime->tm_isdst) +
-				(float)currenttime->tm_min / 60.0f;
-
+				(float)currenttime->tm_sec/3600.0000f;
+	
+	/* Animation */
+            es->rotation += es->rotspeed;
```
Comment (soreau) [inline src/earth/earth.c]: Everything looks good except this should *not* be indented:
`es->rotation += es->rotspeed;`

Comment (Eignar17) [reply src/earth/earth.c]: ok

Comment (soreau) [reply src/earth/earth.c]: Looks good to me, thanks.

**Pull request merged**